File: archive/o/octave/capitalize.m
function capitalize()
    usage = 'Usage: please provide a string';
    arg_list = argv();
    nargin = length(arg_list);
    if nargin == 0
        disp(usage);
        return;
    end

    string = arg_list{1};
    if length(string) == 0
        disp(usage);
        return;
    end

    disp(strcat(upper(string(1:1)), string(2:end)));
end

File: archive/f/f-sharp/Capitalize.fs
open System

module Capitalizer =
    let private capitalize (input: string) =
        let chars = input.ToCharArray()
        chars.[0] <- Char.ToUpper chars.[0]
        String chars

    let run input = capitalize input |> Ok

module Helpers =
    let private (|NonEmptyString|_|) (s: string) =
        if String.IsNullOrWhiteSpace s then None else Some s

    let parseArgs =
        function
        | [| NonEmptyString input |] -> Ok input
        | _ -> Error "Usage: please provide a string"

    let handleResult =
        function
        | Ok result ->
            printfn "%s" result
            0
        | Error msg ->
            eprintfn "%s" msg
            1

[<EntryPoint>]
let main argv =
    argv |> Helpers.parseArgs |> Result.bind Capitalizer.run |> Helpers.handleResult
